extension/scripts/features/hide-stock-blocks/ttHideStockBlocks.js

```diff
@@ -0,0 +1,32 @@
+"use strict";
+
+(async () => {
+	if (!getPageStatus().access) return;
+
+	featureManager.registerFeature(
+		"Hide Stock Blocks",
+		"stocks",
+		() => settings.hideStockBlocks.length,
+		null,
+		hideStocks,
+		unhideStocks,
+		{
+			storage: ["settings.hideStockBlocks.length"],
+		},
+		null
+	);
+
+	async function hideStocks() {
+		await requireElement("#stockmarketroot [class*='stock___'][id]");
+		document.findAll("#stockmarketroot [class*='stock___'][id]").forEach(stockNode => {
+			if (settings.hideStockBlocks.some(x => x == stockNode.getAttribute("id"))) stockNode.classList.add("hidden");
+		});
+		if (!document.find(".tt-stocks-hidden")) document.find("#stockmarketroot [class*='appHeaderWrapper__']").insertAdjacentElement("afterend", await createMessageBox("Some stock blocks have been hidden.", { class: "tt-stocks-hidden" }));
```

Comment:
Due to live reload, you should either remove the message to then add it, or adjust it.